Client/Frontend/Home/PanelDataObservers.swift

```diff
@@ -48,28 +49,33 @@ class ActivityStreamDataObserver: DataObserver {
         events.forEach { NotificationCenter.default.removeObserver(self, name: $0, object: nil) }
     }
     
-    func invalidate(highlights: Bool) {
+    func refresh(highlights: Bool) {
+        guard !profile.isShutdown else {
+            return
+        }
         self.delegate?.willInvalidateDataSources()
 
-        let notify = {
+        // Highlights are cached for 15 mins
+        let invalidateHighlights = highlights ? true : (Date.now() - lastInvalidation > invalidationTime)
```

Comment:
Out of a million users, 1,700 will have a clock that's persistently wrong by more than a minute. If I read your code right, you always invalidate after launch, and then every fifteen minutes thereafter, right? If so, just substitute `uptimeMillis` for `Date.now()` everywhere.